yosh/shell.py

```diff
@@ -16,8 +16,55 @@
 def tokenize(string):
```

Comment:
This function is really messy and hard to read and understand at first glance.
- It contains a pyramid of doom (several nested `if`). A rule of thumb is it should be no more than 2-3 nested levels.
- Meaningless variable name.
- `el.find('=')` is used repeatedly. You can declare a variable name and reuse it.
- Some logic is unnecessary. Ex. 
``` python
if el.find('=') != -1:
    if int(el.find('=')) > 0:
        if int(el.find('=')) != len(token[i]):
```
- First, .find() returns `int`. So, we don't need to cast it with `int()`.
- Second, the second `if` (greater than 0) is guaranteed to not equal `-1`. So, the first `if` check is not necessary.
- Third, `el` and `token[i]` in the loop is the same value. So, we don't need to use two distinct variables.
- Fourth, `el.find('=') != len(token[i])` is always not equal because .find() returns the index which in range [0, len - 1]. So, we don't need to check.